std/test/__init__.py

```diff
@@ -0,0 +1,619 @@
+
+class SpendResult:
+    def __init__(self,result):
+        """Constructor for internal use.
+
+        error - a string describing the error or None
+        result - the raw result from Network::push_tx
+        outputs - a list of new Coin objects surviving the transaction
+        """
+        self.result = result
+        if 'error' in result:
+            self.error = result['error']
+            self.outputs = []
+        else:
+            self.error = None
+            self.outputs = result['additions']
+
+    def find_standard_coins(self,puzzle_hash):
+        """Given a Wallet's puzzle_hash, find standard coins usable by it.
+
+        These coins are recognized as changing the Wallet's chia balance and are
+        usable for any purpose."""
+        return list(filter(lambda x: x.puzzle_hash == puzzle_hash, self.outputs))
+
+class CoinWrapper(Coin):
+    """A class that provides some useful methods on coins."""
+    def __init__(self, parent : Coin, puzzle_hash : bytes32, amt : uint64, source : Program):
+        """Given parent, puzzle_hash and amount, give an object representing the coin"""
+        super().__init__(parent,puzzle_hash,amt)
+        self.source = source
+
+    def puzzle(self) -> Program:
+        """Return the program that unlocks this coin"""
+        return self.source
+
+    def puzzle_hash(self) -> bytes32:
+        """Return this coin's puzzle hash"""
+        return self.puzzle().get_tree_hash()
+
+    def contract(self):
+        """Return a contract object wrapping this coin's program"""
+        return ContractWrapper(DEFAULT_CONSTANTS.GENESIS_CHALLENGE, self.source)
+
+    def create_standard_spend(self, priv, conditions):
+        delegated_puzzle_solution = Program.to((1, conditions))
+        solution = Program.to([[], delegated_puzzle_solution, []])
+
+        coin_solution_object = CoinSolution(
+            self,
+            self.puzzle(),
+            solution,
+        )
+
+        # Create a signature for each of these.  We'll aggregate them at the end.
+        signature = AugSchemeMPL.sign(
+            calculate_synthetic_secret_key(priv, DEFAULT_HIDDEN_PUZZLE_HASH),
+            (delegated_puzzle_solution.get_tree_hash() + self.name() + DEFAULT_CONSTANTS.AGG_SIG_ME_ADDITIONAL_DATA)
+        )
+
+        return coin_solution_object, signature
+
+# We have two cases for coins:
+# - Wallet coins which contribute to the "wallet balance" of the user.
+#   They enable a user to "spend money" and "take actions on the network"
+#   that have monetary value.
+#
+# - Contract coins which either lock value or embody information and
+#   services.  These also contain a chia balance but are used for purposes
+#   other than a fungible, liquid, spendable resource.  They should not show
+#   up in a "wallet" in the same way.  We should use them by locking value
+#   into wallet coins.  We should ensure that value contained in a contract
+#   coin is never destroyed.
+class ContractWrapper:
```

Comment:
It might not be needed but my intention was to encapsulate chialisp code that could potentially be used for a coin from a Coin on the blockchain.  More advanced code would put a curry method on there that verified that the object wasn't over-applied, would support generating the arguments for coin spends and some other things.
I'm not going as fast as I had hoped at this so it's a little bare.